quickwit-indexing/src/actors/packager.rs

```diff
@@ -48,20 +49,21 @@ use crate::models::{
 pub struct Packager {
     actor_name: &'static str,
     uploader_mailbox: Mailbox<PackagedSplitBatch>,
-    /// The special field for extracting tags.
-    tags_field: Field,
+    /// A list of tag fields specified in the index config and their
+    /// corresponding schema field.
+    tag_fields_list: Vec<(String, Field)>,
 }
 
 impl Packager {
     pub fn new(
         actor_name: &'static str,
-        tags_field: Field,
+        tag_fields_list: Vec<(String, Field)>,
```

Comment:
I would introduce a struct `NamedField` struct to make it clearer

Reply:
with `tag_fields` variable

Reply:
```rust
pub fn new(
    ...
    tag_fields: Vec<NamedField>,
    ...
)

```

Reply:
it's ok if the docs is better as I suggested above.

Reply:
I went with NamedField, This is not the first time we need something like this.